### PR TITLE
Fix songs that are entirely ALL CAPS.

### PR DESCRIPTION
**Pull request by bohning**
Lyrics in ALL CAPS is considered as shouting and therefore discouraged/bad style. Thus, if a song’s lyrics are entirely (!) ALL CAPS, they are all changed to lowercase and then the first word (more precisely the first capitalizable letter of the first word) of each line is capitalized again.

The only downside is that names/nouns that should be capitalized are then lowercase, but this should only concern a handful of words, so fixing this manually only takes a small effort.

Good test cases are e.g.:
http://usdb.animux.de/index.php?link=detail&id=9417
http://usdb.animux.de/index.php?link=detail&id=18539
http://usdb.animux.de/index.php?link=detail&id=21467
http://usdb.animux.de/index.php?link=detail&id=22046

### CODE REVIEW

File: src/usdb_syncer/notes_parser.py [hunk 1]
```diff
@@ -300,6 +303,28 @@ def fix_spaces(self) -> None:
...
+                note.text = note.text.lower()
+            self.fix_first_words_capitalization()
+
+    def fix_first_words_capitalization(self) -> None:
+        for line in self.all_lines():
+            # capitalize first capitalizable character
+            # e.g. "“¿que horas son?”" -> "“¿Que horas son?”"
+            text = ""
+            for char_index, char in enumerate(line.notes[0].text):
+                if char == char.capitalize():
```
Comment (RumovZ) [inline src/usdb_syncer/notes_parser.py:318]: Slightly simpler:
```suggestion
                if char != char.upper():
                    line.notes[0].text = line.notes[0].text.replace(char, char.upper(), 1)
                    break
```

File: src/usdb_syncer/notes_parser.py [hunk 2]
```diff
@@ -300,6 +303,28 @@ def fix_spaces(self) -> None:
             # used to be incomplete in USDX, and it allows simple text concatenation
             line.notes[-1].right_trim_text_and_add_space()
 
+    def fix_all_caps(self) -> None:
+        if self.lyrics().isupper():
```
Comment (RumovZ) [inline src/usdb_syncer/notes_parser.py:307]: You don't need to construct a new string here:
```suggestion
        if all(note.text.isupper() for note in self.all_notes()):
```

**Comment by bohning** [comment]
Thanks for the improvement suggestions!

**Comment by RumovZ** [comment]
Sorry, that wasn't meant to be applied verbatim. There are other lines that need to be deleted now.

**Comment by bohning** [comment]
Oh, I see. The convenience to commit the suggested improvements with a single click was too tempting =)

**Comment by bohning** [comment]
Problem is: this code fails on the given song, because the very last syllable is (erroneously) on a single `!`. `isupper()` returns False on the exclamation mark, therefore the `if all()` expression returns False and thus the code is never executed.

**Comment by RumovZ** [comment]
Oh well, I guess case is more complex than it may seem at a first glance. In that case, I would check individual letters:
```py
def is_all_caps(self) -> bool:
    return not any(char.islower() for note in self.all_notes() for char in note.text)
```

**Review by RumovZ — APPROVED** [review]

**Pull request merged**